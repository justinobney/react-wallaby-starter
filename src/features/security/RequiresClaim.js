import {connect} from 'react-redux';
import types from 'prop-types';

export const RequiresClaim = ({claim, claims = [], children}) => {
  if (claims.indexOf(claim) === -1) {
    return null;
  }

  return children || null;
};

RequiresClaim.propTypes = {
  claims: types.array,
  claim: types.string.isRequired,
};

const mapStateToProps = state => {
  return {
    claims: state.identity.claims,
  };
};

export default connect(mapStateToProps)(RequiresClaim);
