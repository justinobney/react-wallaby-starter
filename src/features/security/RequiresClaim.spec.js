import React from 'react';
import {render} from 'react-testing-library';
import {RequiresClaim} from './RequiresClaim';

const ChildComponent = () => <div>Child Component</div>;

describe('Component: RequiresClaim', () => {
  it('should render component when user has claim', () => {
    const {container} = render(
      <RequiresClaim claim="users/modify" claims={['users/modify']}>
        Testing
      </RequiresClaim>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render null when user does not have claim', () => {
    const {container} = render(
      <RequiresClaim
        claim="users/modify"
        claims={[]}
        render={() => <ChildComponent />}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
