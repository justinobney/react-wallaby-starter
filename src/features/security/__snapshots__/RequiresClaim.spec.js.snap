// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: RequiresClaim should render component when user has claim 1`] = `
<div>
  Child Component
</div>
`;

exports[`Component: RequiresClaim should render null when user does not have claim 1`] = `null`;
